File: Data Structures & Algorithms/jump-game/submission-2.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        target_index = n - 1

        for i in reversed(range(n-1)):
            if nums[i] >= target_index - i:
                target_index = i



        return target_index == 0

File: Data Structures & Algorithms/missing-number/submission-2.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum = 0;
        int act_sum = 0;
        for (int i=0; i<nums.size(); i++) {
            sum += i;
            act_sum += nums[i];
        }
        sum += nums.size();

        return sum - act_sum;
    }
};
